device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/fileupload/FileUploadTask.java

```diff
@@ -144,7 +145,6 @@ public void run()
         catch (StorageException | IOException | IllegalArgumentException | URISyntaxException e)
         {
             log.error("File upload failed to upload the stream to the blob", e);
-            fileUploadCompletionNotification = new FileUploadCompletionNotification(sasUriResponse.getCorrelationId(), false, -1, "Failed to upload to storage.");
```

Comment:
Moving this default value assignment to earlier so that the finally block always has a non-null instance to use